src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3715,6 +3905,11 @@ private void ReinferMethodAndVisitArguments(BoundCall node, TypeWithState receiv
             (method, results, returnNotNull) = VisitArguments(node, node.Arguments, refKindsOpt, method.Parameters, node.ArgsToParamsOpt,
                 node.Expanded, node.InvokedAsExtensionMethod, method);
 
+            if (method.IsStatic || node.ReceiverOpt is BoundThisReference || node.ReceiverOpt is BoundBaseReference)
```

Comment:
>method.IsStatic || node.ReceiverOpt is BoundThisReference || node.ReceiverOpt is BoundBaseReference [](start = 16, length = 99)

Why would we do this only for statics and this/base calls?  Is it done elsewhere for other calls?

Reply:
Looking back at the LDM notes, I think you're right. 
We intend to support other scenarios like `x.HasValue` followed by `x.Value` for instance. I'll relax this restriction.